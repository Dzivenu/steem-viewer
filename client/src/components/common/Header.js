import React, { Component } from 'react';
import styled from 'styled-components';
import EN from '../../constants/translations/en';
import { COLORS } from '../../constants/styles';

const Container = styled.header`
  background-color: ${COLORS.BLUE.CORNFLOWER_BLUE};
  box-shadow: 0 2px 4px 0 rgba(0,0,0,.05);
  color: ${COLORS.WHITE.WHITE};
  margin-bottom: 1em;
  padding: 1em;
  position: fixed;
  width: 100%;
`;

class Header extends Component {
  render() {
    return <Container>{EN.title}</Container>;
  }
}

export default Header;
