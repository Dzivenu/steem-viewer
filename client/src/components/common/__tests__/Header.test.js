import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../Header';

Enzyme.configure({adapter: new Adapter()});

describe('(Component) Header', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {};
      const wrapper = shallow(<Header {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
