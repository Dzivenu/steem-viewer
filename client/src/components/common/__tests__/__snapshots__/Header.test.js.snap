// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Header with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": "Steem Viewer",
    },
    "ref": null,
    "rendered": "Steem Viewer",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "Steem Viewer",
      },
      "ref": null,
      "rendered": "Steem Viewer",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
